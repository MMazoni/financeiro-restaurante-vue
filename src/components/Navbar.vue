<template>
  <nav>
    <v-app-bar flat app>
      <v-app-bar-nav-icon class="grey--text" @click="drawer = !drawer">
        <v-icon>menu</v-icon>
      </v-app-bar-nav-icon>
      <v-toolbar-title class="text-uppercase grey--text">
        <span class="font-weight-light">Finen</span>
        <span>Place</span>
      </v-toolbar-title>
      <v-spacer></v-spacer>
      <v-btn text color="grey">
        <span>Sair</span>
        <v-icon right>exit_to_app</v-icon>
      </v-btn>
    </v-app-bar>

    <v-navigation-drawer v-model="drawer" app class="primary lighten-1">
      <v-layout column align-center>
        <v-flex class="mt-5">
          <v-avatar size="100">
            <img src="https://avatars1.githubusercontent.com/u/37179593?s=460&u=49c29fed6531de1a1e09fed0ff13d2582033d060&v=4" alt="UsuÃ¡rio Avatar">
          </v-avatar>
        </v-flex>
        <p class="dark_gray--text subheading mt-1">Matheus Mazoni</p>
      </v-layout>
      <v-list>
        <v-list-item v-for="link in links" :key="link.text" router :to="link.route">
          <v-list-item-action>
            <v-icon>{{ link.icon }}</v-icon>
          </v-list-item-action>
          <v-list-item-content>
            <v-list-item-title class="dark_gray--text">{{ link.text }}</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list>
    </v-navigation-drawer>
  </nav>
</template>

<script>
export default {
  name: "Navbar",
  data() {
    return {
      drawer: false,
      links: [
        {icon: 'dashboard', text: 'Dashboard', route: '/'},
        {icon: 'local_atm', text: 'Caixa', route: '/caixa'},
        {icon: 'local_shipping', text: 'Fornecedor', route: '/fornecedor'},
        {icon: 'receipt', text: 'Nota Fiscal', route: '/nota-fiscal'},
        {icon: 'inventory_2', text: 'Produto', route: '#'},
        {icon: 'person', text: 'Equipe', route: '/equipe'},
      ]
    };
  },
};
</script>