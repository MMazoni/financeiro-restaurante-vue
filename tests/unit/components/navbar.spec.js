import { RouterLinkStub, mount } from '@vue/test-utils';
import Navbar from '@/components/Navbar.vue';
import Vue from 'vue';
import Vuetify from 'vuetify';

const vuetify = new Vuetify();

Vue.use(Vuetify);
describe('Navbar.vue', () => {

  it('should snapshot', () => {
    const wrapper = mount(Navbar, {
      vuetify,
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('is a Vue instance', () => {
    const wrapper = mount(Navbar, {
      vuetify,
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.vm).toBeTruthy();
  });
});
