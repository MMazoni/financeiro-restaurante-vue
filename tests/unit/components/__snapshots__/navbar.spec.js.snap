// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar.vue should snapshot 1`] = `
<nav>
  <header class="v-sheet theme--light v-toolbar v-toolbar--flat v-app-bar v-app-bar--fixed" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
    <div class="v-toolbar__content" style="height: 64px;"><button type="button" class="v-app-bar__nav-icon grey--text v-btn v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">menu</i></span></button>
      <div class="v-toolbar__title text-uppercase grey--text"><span class="font-weight-light">Finen</span> <span>Place</span></div>
      <div class="spacer"></div> <button type="button" class="v-btn v-btn--text theme--light v-size--default grey--text"><span class="v-btn__content"><span>Sair</span> <i aria-hidden="true" class="v-icon notranslate v-icon--right material-icons theme--light">exit_to_app</i></span></button>
    </div>
  </header>
  <nav class="primary lighten-1 v-navigation-drawer v-navigation-drawer--close v-navigation-drawer--fixed v-navigation-drawer--is-mobile theme--light" style="height: 100vh; top: 0px; transform: translateX(-100%); width: 256px;">
    <div class="v-navigation-drawer__content">
      <div class="layout column align-center">
        <div class="flex mt-5">
          <div class="v-avatar" style="height: 100px; min-width: 100px; width: 100px;"><img src="https://avatars1.githubusercontent.com/u/37179593?s=460&amp;u=49c29fed6531de1a1e09fed0ff13d2582033d060&amp;v=4" alt="UsuÃ¡rio Avatar"></div>
        </div>
        <p class="dark_gray--text subheading mt-1">Matheus Mazoni</p>
      </div>
      <div class="v-list v-sheet theme--light"><a tabindex="0" router="" class="v-list-item v-list-item--link theme--light">
          <div class="v-list-item__action"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">dashboard</i></div>
          <div class="v-list-item__content">
            <div class="v-list-item__title dark_gray--text">Dashboard</div>
          </div>
        </a><a tabindex="0" router="" class="v-list-item v-list-item--link theme--light">
          <div class="v-list-item__action"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">local_atm</i></div>
          <div class="v-list-item__content">
            <div class="v-list-item__title dark_gray--text">Caixa</div>
          </div>
        </a><a tabindex="0" router="" class="v-list-item v-list-item--link theme--light">
          <div class="v-list-item__action"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">person</i></div>
          <div class="v-list-item__content">
            <div class="v-list-item__title dark_gray--text">Equipe</div>
          </div>
        </a></div>
    </div>
    <div class="v-navigation-drawer__border"></div>
  </nav>
</nav>
`;
